### PR TITLE
post changes

### PR DESCRIPTION
**Pull request by anikarabenhorst**

### CODE REVIEW

File: src/components/PostList/Post.js [hunk 1]
```diff
@@ -22,6 +25,41 @@ const Post = ({post}) => {
     const postedBy = useSelector((state) =>
         state.allUsers.find((user) => user?._id === post.userId)
     );
+    const [liked, setLiked] = useState(post.liked.includes(currentUser?._id));
+    const [disliked, setDisliked] = useState(post.disliked.includes(currentUser?._id));
```
Comment (anikarabenhorst) [inline src/components/PostList/Post.js]: issue is with these includes to setState -> fine initially but not on update.

File: src/components/CommentModal.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        onClose={onClose}
+        size={"xl"}
+      >
+        <ModalOverlay />
+        <ModalContent align={"center"}>
+          <ModalCloseButton />
+          <ModalBody pb={6}>
+            <Heading size={"xl"} color={"teal"}>{post.song_title}</Heading>
+            <Heading size={"l"}>Comments</Heading>
+            {post.comments.map((comment) =>
```
Comment (rkalafos) [inline src/components/CommentModal.js]: Is there a default empty list for when there are no comments?

Comment (anikarabenhorst) [reply src/components/CommentModal.js]: yes each post is created with `[] `for comments

File: src/components/PostList/Post.js (hunk 1 above)
Comment (rkalafos) [reply src/components/PostList/Post.js]: Yeah I haven't seen people use set state like this, not saying its impossible but I haven't tried it. I don't think you need to do uesState, you can just do 
```
const liked = post.liked.includes(currentUser?._id);
```
And then customize your own on liked function that will update the state.

File: src/components/PostList/Post.js [hunk 3]
```diff
@@ -10,21 +12,78 @@ import {
...
+    const { isOpen, onOpen, onClose } = useDisclosure();
     const postedBy = useSelector((state) =>
         state.allUsers.find((user) => user?._id === post.userId)
     );
+    const [liked, setLiked] = useState(post.liked.includes(currentUser?._id));
+    const [disliked, setDisliked] = useState(post.disliked.includes(currentUser?._id));
+
+
+    const onClickLikePost = (e) => {
+        e.preventDefault();
```
Comment (rkalafos) [inline src/components/PostList/Post.js]: Is the issue that this function isn't working or that the `liked` variable isn't updating?

File: src/components/PostList/index.js [hunk 4]
```diff
@@ -10,13 +10,16 @@ function filterPosts(posts, user) {
   if (Object.keys(posts).length === 0) {
     return [];
   } // If the user is not logged in, return the 5 latest posts
-  else if (Object.keys(user).length === 0) {
```
Comment (rkalafos) [inline src/components/PostList/index.js:13]: Ah yes thanks. What about if route is "/"? Is that any different?

File: src/components/PostList/Post.js (hunk 3 above)
Comment (anikarabenhorst) [reply src/components/PostList/Post.js]: it doesn't get to the backend AND there's the includes problem. As soon as i press the like it fails

**Pull request merged**